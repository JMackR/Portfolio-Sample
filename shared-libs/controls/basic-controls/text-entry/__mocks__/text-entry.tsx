import React from 'react'

export const TextEntry = React.forwardRef(() => <div>Text Entry</div>)
