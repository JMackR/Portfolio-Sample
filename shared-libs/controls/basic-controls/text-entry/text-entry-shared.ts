import { useColorsForTextColorsCollection, useFontForTextType } from '@sample/themes'
import type { TextEntryProps } from './text-entry.props'

export const ComputePropsForTextEntryProps = (props: TextEntryProps) => {
  const {
    autoCapitalize,
    autoCorrect,
    autoCompleteType,
    textType,
    textColor: textColorName,
    hintColor: hintColorName,
    linesMaximum,
    linesMinimum,
    tintColor: tintColorName,
  } = props

  const font = useFontForTextType(textType !== undefined ? textType : 'secondary3')
  const minLineHeight = Math.max(Math.floor(linesMinimum !== undefined ? linesMinimum : 1), 1)
  const maxLineHeight = Math.max(Math.floor(linesMaximum !== undefined ? linesMaximum : minLineHeight), minLineHeight)
  const isMultiline = minLineHeight > 1 || maxLineHeight > 1
  const minPointHeight = minLineHeight * font.lineHeight
  const maxPointHeight = maxLineHeight * font.lineHeight

  const [primaryColor, hintColor, tintColor] = useColorsForTextColorsCollection([
    textColorName !== undefined ? textColorName : 'primary3',
    hintColorName !== undefined ? hintColorName : 'primary2',
    tintColorName !== undefined ? tintColorName : 'primary1',
  ])

  return {
    autoCapitalize: autoCapitalize !== undefined ? autoCapitalize : 'none',
    autoCorrect: autoCorrect !== undefined ? autoCorrect : true,
    autoCompleteType: autoCompleteType !== undefined ? autoCompleteType : 'off', // Android only
    font,
    minLineHeight,
    maxLineHeight,
    minPointHeight,
    maxPointHeight,
    isMultiline,
    primaryColor,
    hintColor,
    tintColor,
  }
}
