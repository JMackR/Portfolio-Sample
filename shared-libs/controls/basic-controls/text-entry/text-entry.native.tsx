import React, { useState, forwardRef, useImperativeHandle, useRef } from 'react'
import {
  StyleSheet,
  TextInput,
  NativeSyntheticEvent,
  TextInputEndEditingEventData,
  InputAccessoryView,
  Platform,
  Button,
} from 'react-native'
import { BackgroundContainer, Flex, Margin, Separator } from '@sample/layout-controls'
import { ComputePropsForTextEntryProps } from './text-entry-shared'
import { TextEntryProps, TextEntryRef } from './text-entry.props'
import { BottomSheetTextInput } from '@gorhom/bottom-sheet'

const isIOS = Platform.OS === 'ios'
const STICKY_KEYBOARD_BAR_HEIGHT = 40
const RIGHT_MARGIN = 4
const inputAccessoryViewID = 'stickyDoneBar'

export const TextEntry = forwardRef<TextEntryRef, TextEntryProps>((props, ref) => {
  const {
    text,
    hint,
    inputCharLimit,
    returnKeyType,
    keyboardType,
    secureTextEntry,
    textAlign,
    showSoftInputOnFocus = true,
    keyPressHandler,
    textChangeHandler,
    endEditingHandler,
    blurHandler,
    focusHandler,
    onSubmitEditing,
    testID,
    showInputAccessoryView,
    returnKeyLabel,
    onPressIn,
    modalInput,
    styles: stylesCustom,
  } = props
  const {
    autoCapitalize,
    autoCorrect,
    autoCompleteType,
    font,
    minPointHeight,
    maxPointHeight,
    isMultiline,
    primaryColor,
    hintColor,
    tintColor,
  } = ComputePropsForTextEntryProps(props)

  const styles = StyleSheet.create({
    textentry: {
      flex: 0,
      flexDirection: 'row',
      color: tintColor,
      fontSize: font.fontSize,
      fontWeight: font.fontWeight,
      fontFamily: font.fontFamily,
      padding: 0,
      margin: 0,
      minHeight: minPointHeight,
      maxHeight: maxPointHeight,
      alignSelf: 'stretch',
      textAlign,
      textAlignVertical: isMultiline ? 'top' : 'auto',
      ...stylesCustom?.container,
    },
  })

  const [, setHeight] = useState(minPointHeight)
  // const { keyboardAvoidanceFocus, keyboardAvoidanceBlur, generateFocusId } = useKeyboardAvoidanceFocus()
  // const focusIdRef = useRef(generateFocusId())
  const input = useRef<TextInput>(null)

  const keyPressFunc = (e: { nativeEvent: { key: string } }) => {
    if (keyPressHandler !== undefined) {
      keyPressHandler(e.nativeEvent.key)
    }
  }

  const endEditingFunc = (e: NativeSyntheticEvent<TextInputEndEditingEventData>) => {
    if (endEditingHandler !== undefined) {
      endEditingHandler(e.nativeEvent.text)
    }
  }
  const handleSubmitEditing = (e: NativeSyntheticEvent<TextInputEndEditingEventData>) => {
    if (onSubmitEditing) {
      onSubmitEditing(e.nativeEvent.text)
    }
  }

  const contentSizeChangeFunc = (e: { nativeEvent: { contentSize: { width: number; height: number } } }) => {
    setHeight(Math.min(e.nativeEvent.contentSize.height, maxPointHeight))
  }

  const focusFunc = () => {
    // keyboardAvoidanceFocus(focusIdRef.current)

    focusHandler && focusHandler()
  }

  const blurFunc = () => {
    // keyboardAvoidanceBlur(focusIdRef.current)
    blurHandler && blurHandler()
  }

  const focus = () => {
    if (input.current !== null) {
      input.current.focus()
    }
  }

  const blur = () => {
    if (input.current !== null) {
      input.current.blur()
    }
  }

  const setPrivate = () => {
    // if (input.current !== null) {
    //   Instabug.setPrivateView(input.current)
    // }
  }

  useImperativeHandle(ref, () => ({
    focus,
    blur,
    setPrivate,
  }))
  const Input = modalInput ? BottomSheetTextInput : TextInput
  return (
    <>
      <Input
        style={styles.textentry}
        ref={input}
        underlineColorAndroid={'transparent'}
        autoCapitalize={autoCapitalize}
        autoCompleteType={autoCompleteType}
        autoCorrect={autoCorrect}
        value={text}
        onPressIn={onPressIn}
        maxLength={inputCharLimit}
        multiline={isMultiline}
        onKeyPress={keyPressFunc}
        onEndEditing={endEditingFunc}
        onFocus={focusFunc}
        onBlur={blurFunc}
        onSubmitEditing={handleSubmitEditing}
        keyboardType={keyboardType}
        returnKeyType={returnKeyType}
        secureTextEntry={secureTextEntry}
        showSoftInputOnFocus={showSoftInputOnFocus}
        placeholder={hint}
        returnKeyLabel={returnKeyLabel}
        placeholderTextColor={hintColor}
        onChangeText={textChangeHandler}
        onContentSizeChange={contentSizeChangeFunc}
        selectionColor={tintColor}
        testID={testID || 'text-entry'}
        accessibilityLabel={testID || 'text-entry'}
        inputAccessoryViewID={inputAccessoryViewID}
      />
      {isIOS && showInputAccessoryView && (
        <InputAccessoryView nativeID={inputAccessoryViewID}>
          <BackgroundContainer type={'background4'} isOverlay />
          <Separator />
          <Flex height={STICKY_KEYBOARD_BAR_HEIGHT} crossAxisDistribution={'center'} axisDistribution={'flex-end'}>
            <Margin marginRightStep={RIGHT_MARGIN}>
              <Button onPress={blur} title={'done'} />
            </Margin>
          </Flex>
        </InputAccessoryView>
      )}
    </>
  )
})
