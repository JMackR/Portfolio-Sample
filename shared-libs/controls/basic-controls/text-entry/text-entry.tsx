import React, { useRef, KeyboardEvent, forwardRef, useImperativeHandle, useState } from 'react'
import { useColor, useColorForBackgroundColor, useColorForTextColor } from '@sample/themes'
import { convertFontWeight } from '@sample/themes/utility/font-conversions'
import { StyleSheet, css } from 'aphrodite/no-important'
import { Text } from '../text/text'
import { ComputePropsForTextEntryProps } from './text-entry-shared'
import { TextEntryProps, TextEntryRef } from './text-entry.props'

export const TextEntry = forwardRef<TextEntryRef, TextEntryProps>((props, ref) => {
  const {
    controlName,
    text,
    hint,
    inputCharLimit,
    keyboardType,
    inputMode,
    textColor,
    secureTextEntry,
    keyPressHandler,
    textChangeHandler,
    endEditingHandler,
    blurHandler,
    focusHandler,
    textAlign,
    testID,
    inputDisabled,
    onPaste,
  } = props
  const { minLineHeight, maxLineHeight, font, minPointHeight, maxPointHeight, isMultiline, primaryColor, hintColor } =
    ComputePropsForTextEntryProps(props)

  const { colors } = useColor()
  const styles = StyleSheet.create({
    textentry: {
      '::placeholder': {
        color: hintColor || useColorForTextColor('primary2'),
      },
      ':disabled': {
        color: useColorForTextColor('primary2'),
        '-webkit-text-fill-color': useColorForTextColor('primary2'),
        background: colors.grey100,
      },
      'text-align': textAlign,
      color: textColor || primaryColor,
      backgroundColor: useColorForBackgroundColor('background1'),
      overflow: 'hidden',
      resize: 'none',
      minWidth: 0,
      minSize: 0,
      textVerticalAlign: 'top',
      // TODO: Check back after font changes
      // minHeight: `${minPointHeight}px`,
      // minHeight: '22px',
      height: '16px',
      fontSize: font.fontSize,
      fontWeight: convertFontWeight(font.fontWeight),
      fontFamily: `${font.fontFamily}, sans-serif`,
      margin: 0,
      display: 'flex',
      flexDirection: 'row',
      width: 'calc(100% - 30px)',
      borderRadius: '4px',
      border: 'none',
      padding: '20px 16px 20px 14px',
      outline: 'none',
      boxShadow: 'none',
    },

    inputContainer: {
      position: 'relative',
      display: 'flex',
      flexDirection: 'column',
      ':focus-within .inputTransform': {
        top: '0',
        left: '14px',
        fontSize: '10px',
        transform: 'translateY(0px) scale(0.8)',
        transformOrigin: '0px 100%',
      },
    },
    inputLabel: {
      position: 'absolute',
      pointerEvents: 'none',
      transform: 'translateY(-50%)',
      transformOrigin: '0px 100%',
      transition: 'all 0.15s ease-out 0s',
      top: '50%',
      left: '14px',
    },

    inputFilled: {
      top: '0',
      left: '14px',
      fontSize: '10px',
      transform: 'translateY(0px) scale(0.8)',
      transformOrigin: '0px 100%',
    },
  })
  const [isActive, setIsActive] = useState(false)

  const input = isMultiline ? useRef<HTMLTextAreaElement>() : useRef<HTMLInputElement>()

  const keyPressFunc = (e: KeyboardEvent<HTMLTextAreaElement> | KeyboardEvent<HTMLInputElement>) => {
    const element = input.current
    if (e.key === 'Enter' && isMultiline === false && element !== undefined && element !== null) {
      element.blur()
    }

    keyPressHandler && keyPressHandler(e.key)
  }

  const changeFunc = () => {
    const element = input.current
    if (element !== undefined && element !== null && isMultiline && minLineHeight !== maxLineHeight) {
      element.style.height = 'auto'
      element.style.height = `${Math.min(element.scrollHeight, maxPointHeight)}px`
    }

    if (textChangeHandler) {
      textChangeHandler(element !== undefined && element !== null ? element.value : undefined)
    }
  }

  const blurFunc = () => {
    if (endEditingHandler !== undefined) {
      const element = input.current
      endEditingHandler(element !== undefined && element !== null ? element.value : '')
    }
    if (blurHandler !== undefined) {
      setIsActive(false)
      blurHandler()
    }
  }

  const focus = () => {
    if (input.current !== undefined && input.current !== null) {
      input.current.focus()
      setIsActive(true)
    }
  }

  const blur = () => {
    if (input.current !== undefined && input.current !== null) {
      input.current.blur()
    }
  }

  const setPrivate = () => {}

  useImperativeHandle(ref, () => ({
    focus,
    blur,
    setPrivate,
  }))

  if (isMultiline) {
    return (
      <textarea
        className={css(styles.textentry)}
        name={controlName}
        maxLength={inputCharLimit}
        onKeyPress={keyPressFunc}
        onFocus={focusHandler}
        onBlur={blurFunc}
        ref={input as React.RefObject<HTMLTextAreaElement>}
        placeholder={hint}
        onChange={changeFunc}
        value={text}
      />
    )
  }

  return (
    <div className={css(styles.inputContainer)}>
      <input
        className={css(styles.textentry)}
        name={controlName}
        maxLength={inputCharLimit}
        onFocus={focusHandler}
        onBlur={blurFunc}
        onKeyDown={keyPressFunc}
        type={secureTextEntry ? 'password' : keyboardType}
        inputMode={inputMode}
        ref={input as React.RefObject<HTMLInputElement>}
        // placeholder={hint}
        onChange={changeFunc}
        value={text || ''}
        data-testid={testID}
        disabled={inputDisabled}
        onPaste={onPaste}
      />
      <label
        className={
          text
            ? `${css(styles.inputLabel)} ${css(styles.inputFilled)} inputTransform`
            : `${css(styles.inputLabel)} inputTransform`
        }
        htmlFor={controlName}
      >
        <Text textType={'secondary4'} color={inputDisabled ? 'primary2' : 'primary1'}>
          {hint}
        </Text>
      </label>
    </div>
  )
})
