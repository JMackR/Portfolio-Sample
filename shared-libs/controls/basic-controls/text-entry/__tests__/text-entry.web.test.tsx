import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { StyleSheetTestUtils } from 'aphrodite/no-important'
import { createSerializer } from 'jest-aphrodite-react/no-important'
import renderer from 'react-test-renderer'
import { TextEntry } from '../text-entry'
import { TextEntryProps } from '../text-entry.props'

expect.addSnapshotSerializer(createSerializer(() => StyleSheetTestUtils, { removeVendorPrefixes: true }))

describe('Text Entry', () => {
  let props: TextEntryProps
  beforeEach(() => {
    props = {
      hint: 'Test Hint',
    }
  })

  describe('has text', () => {
    beforeEach(() => {
      props = {
        text: '123672',
      }
    })
    it('should render the text', () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      expect(input).toBeDefined()
    })
  })

  describe('has label text', () => {
    beforeEach(() => {
      props = {
        hint: 'Test Hint',
      }
    })
    it('should render the text', () => {
      const { getByText } = render(<TextEntry {...props} />)
      expect(getByText(props.hint as string)).toBeDefined()
    })
  })

  describe('on text change', () => {
    beforeEach(() => {
      props = {
        text: 'Test Text',
        textChangeHandler: jest.fn(),
      }
    })
    it('should call textChangeHandler', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      fireEvent.change(input, { target: { value: 'new value' } })
      await waitFor(() => {
        expect(props.textChangeHandler).toHaveBeenCalledTimes(1)
      })
    })
    it('should call correct value', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      fireEvent.change(input, { target: { value: 'new value' } })
      await waitFor(() => {
        expect(props.textChangeHandler).toHaveBeenCalledWith('new value')
      })
    })
  })

  describe('has keyboard type', () => {
    beforeEach(() => {
      props = {
        text: '561651651',
        keyboardType: 'number-pad',
      }
    })
    it('should have keyboard type as number-pad', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      expect(input).toHaveAttribute('type', 'number-pad')
    })
  })

  describe('has inputMode', () => {
    beforeEach(() => {
      props = {
        text: '135156',
        inputMode: 'numeric',
      }
    })
    it('should have input mode as numeric', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      expect(input).toHaveAttribute('inputmode', 'numeric')
    })
  })

  describe('is multiline', () => {
    beforeEach(() => {
      props = {
        text: '135156',
        inputMode: 'numeric',
        linesMinimum: 3,
      }
    })
    it('should render textarea', async () => {
      const { getByRole } = render(<TextEntry {...props} />)
      const input = getByRole('textbox')
      expect(input).toBeInTheDocument()
    })
  })

  describe('has secureTextEntry', () => {
    beforeEach(() => {
      props = {
        text: 'default text',
        secureTextEntry: true,
      }
    })
    it('should have type as password', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      expect(input).toHaveAttribute('type', 'password')
    })
  })

  describe('on key press', () => {
    beforeEach(() => {
      props = {
        text: 'default text',
        keyPressHandler: jest.fn(),
      }
    })
    it('should call keyPressHandler', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      fireEvent.keyDown(input, { key: 'A', code: 'KeyA' })
      await waitFor(() => {
        expect(props.keyPressHandler).toHaveBeenCalledTimes(1)
      })
    })
  })

  describe('input disabled', () => {
    beforeEach(() => {
      props = {
        text: 'default text',
        inputDisabled: true,
      }
    })
    it('should have disabled true', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      expect(input).toHaveAttribute('disabled')
    })
  })

  describe('on focus', () => {
    beforeEach(() => {
      props = {
        text: 'default text',
        focusHandler: jest.fn(),
      }
    })
    it('should call focusHandler', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      fireEvent.focus(input)
      await waitFor(() => {
        expect(props.focusHandler).toHaveBeenCalledTimes(1)
      })
    })
  })

  describe('on focus out', () => {
    beforeEach(() => {
      props = {
        text: 'default text',
        blurHandler: jest.fn(),
        endEditingHandler: jest.fn(),
      }
    })
    it('should call blurHandler', async () => {
      const { getByDisplayValue } = render(<TextEntry {...props} />)
      const input = getByDisplayValue(props.text as string)
      fireEvent.focusOut(input)
      await waitFor(() => {
        expect(props.blurHandler).toHaveBeenCalledTimes(1)
        expect(props.endEditingHandler).toHaveBeenCalledTimes(1)
      })
    })
  })
})

describe('Text Entry Snapshot Tests', () => {
  it('Text Entry has placeholder renders properly', () => {
    const tree = renderer.create(<TextEntry hint={'Test Hint'} />)
    expect(tree).toMatchSnapshot()
  })
  it('Text Entry is disabled renders properly', () => {
    const tree = renderer.create(<TextEntry inputDisabled={true} />)
    expect(tree).toMatchSnapshot()
  })

  it('Text Color primary renders properly', () => {
    const tree = renderer.create(<TextEntry textColor={'primary1'} />)
    expect(tree).toMatchSnapshot()
  })
})
