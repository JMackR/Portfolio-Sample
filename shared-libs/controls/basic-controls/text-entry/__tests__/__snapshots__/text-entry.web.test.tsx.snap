// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Entry Snapshot Tests Text Color primary renders properly 1`] = `
.inputContainer {
  position: relative;
  display: flex;
  flex-direction: column;
}

.inputContainer:focus-within .inputTransform {
  top: 0;
  left: 14px;
  font-size: 10px;
  transform: translateY(0px) scale(0.8);
  transform-origin: 0px 100%;
}

.inputLabel {
  position: absolute;
  pointer-events: none;
  transform: translateY(-50%);
  transform-origin: 0px 100%;
  transition: all 0.15s ease-out 0s;
  top: 50%;
  left: 14px;
}

.text-textStyle {
  color: #696969;
  font-size: 14px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
  text-decoration-line: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.textentry {
  text-align: undefined;
  color: primary1;
  background-color: #FFFFFF;
  overflow: hidden;
  resize: none;
  min-width: 0px;
  min-size: 0px;
  text-vertical-align: top;
  height: 16px;
  font-size: 16px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  margin: 0px;
  display: flex;
  flex-direction: row;
  width: calc(100% - 30px);
  border-radius: 4px;
  border: none;
  padding: 20px 16px 20px 14px;
  outline: none;
  box-shadow: none;
}

.textentry::placeholder {
  color: #AAAAAF;
}

.textentry:disabled {
  color: #AAAAAF;
  background: #F0F0F0;
}

<div
  className="inputContainer"
>
  <input
    className="textentry"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
  <label
    className="inputLabel inputTransform"
  >
    <div
      className="text-textStyle"
    />
  </label>
</div>
`;

exports[`Text Entry Snapshot Tests Text Entry has placeholder renders properly 1`] = `
.inputContainer {
  position: relative;
  display: flex;
  flex-direction: column;
}

.inputContainer:focus-within .inputTransform {
  top: 0;
  left: 14px;
  font-size: 10px;
  transform: translateY(0px) scale(0.8);
  transform-origin: 0px 100%;
}

.textentry {
  text-align: undefined;
  color: #23231E;
  background-color: #FFFFFF;
  overflow: hidden;
  resize: none;
  min-width: 0px;
  min-size: 0px;
  text-vertical-align: top;
  height: 16px;
  font-size: 16px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  margin: 0px;
  display: flex;
  flex-direction: row;
  width: calc(100% - 30px);
  border-radius: 4px;
  border: none;
  padding: 20px 16px 20px 14px;
  outline: none;
  box-shadow: none;
}

.textentry::placeholder {
  color: #AAAAAF;
}

.textentry:disabled {
  color: #AAAAAF;
  background: #F0F0F0;
}

.inputLabel {
  position: absolute;
  pointer-events: none;
  transform: translateY(-50%);
  transform-origin: 0px 100%;
  transition: all 0.15s ease-out 0s;
  top: 50%;
  left: 14px;
}

.text-textStyle {
  color: #696969;
  font-size: 14px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
  text-decoration-line: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

<div
  className="inputContainer"
>
  <input
    className="textentry"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
  <label
    className="inputLabel inputTransform"
  >
    <div
      className="text-textStyle"
    >
      Test Hint
    </div>
  </label>
</div>
`;

exports[`Text Entry Snapshot Tests Text Entry is disabled renders properly 1`] = `
.inputContainer {
  position: relative;
  display: flex;
  flex-direction: column;
}

.inputContainer:focus-within .inputTransform {
  top: 0;
  left: 14px;
  font-size: 10px;
  transform: translateY(0px) scale(0.8);
  transform-origin: 0px 100%;
}

.textentry {
  text-align: undefined;
  color: #23231E;
  background-color: #FFFFFF;
  overflow: hidden;
  resize: none;
  min-width: 0px;
  min-size: 0px;
  text-vertical-align: top;
  height: 16px;
  font-size: 16px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  margin: 0px;
  display: flex;
  flex-direction: row;
  width: calc(100% - 30px);
  border-radius: 4px;
  border: none;
  padding: 20px 16px 20px 14px;
  outline: none;
  box-shadow: none;
}

.textentry::placeholder {
  color: #AAAAAF;
}

.textentry:disabled {
  color: #AAAAAF;
  background: #F0F0F0;
}

.inputLabel {
  position: absolute;
  pointer-events: none;
  transform: translateY(-50%);
  transform-origin: 0px 100%;
  transition: all 0.15s ease-out 0s;
  top: 50%;
  left: 14px;
}

.text-textStyle {
  color: #AAAAAF;
  font-size: 14px;
  font-weight: 400;
  font-family: ABCSocial-Regular, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
  text-decoration-line: undefined;
}

<div
  className="inputContainer"
>
  <input
    className="textentry"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
  <label
    className="inputLabel inputTransform"
  >
    <div
      className="text-textStyle"
    />
  </label>
</div>
`;
